api/client/joinservice.go

```diff
@@ -125,3 +131,71 @@ func (c *JoinServiceClient) RegisterUsingAzureMethod(ctx context.Context, challe
 	}
 	return certsResp.Certs, nil
 }
+
+// RegisterUsingTPMMethod registers the caller using the TPM join method and
+// returns signed certs to join the cluster. The caller must provide a
+// ChallengeResponseFunc which returns a *proto.RegisterUsingTPMMethodRequest
+// for a given challenge, or an error.
+func (c *JoinServiceClient) RegisterUsingTPMMethod(
```

Comment:
Do we have tests for this in the other PR? Would it make sense to have the client-side tests here?

Reply:
Bump? I think this is the only comment left?

Reply:
[55834c9](https://github.com/gravitational/teleport/pull/40647/commits/55834c9dd7f64f33eabb28ce22dd170310bc21f9)